src/policy/compiler.rs

```diff
@@ -164,38 +169,64 @@ impl Property for CompilerExtData {
             branch_prob: None,
             sat_cost: f64::MAX,
             dissat_cost: Some(0.0),
+            sig_type: PhantomData,
         }
     }
 
     fn from_pk_k() -> Self {
         CompilerExtData {
             branch_prob: None,
-            sat_cost: 73.0,
+            sat_cost: match Ctx::sig_type() {
+                SigType::Ecdsa => 73.,
+                SigType::Schnorr => 64.,
+            },
             dissat_cost: Some(1.0),
+            sig_type: PhantomData,
         }
     }
 
     fn from_pk_h() -> Self {
         CompilerExtData {
             branch_prob: None,
-            sat_cost: 73.0 + 34.0,
-            dissat_cost: Some(1.0 + 34.0),
+            sat_cost: match Ctx::sig_type() {
+                SigType::Ecdsa => 73.0 + 34.0,
+                SigType::Schnorr => 64. + 32.,
+            },
+            dissat_cost: Some(
+                1.0 + match Ctx::sig_type() {
+                    SigType::Ecdsa => 34.0,
+                    SigType::Schnorr => 32.,
+                },
+            ),
+            sig_type: PhantomData,
         }
     }
 
     fn from_multi(k: usize, _n: usize) -> Self {
         CompilerExtData {
             branch_prob: None,
-            sat_cost: 1.0 + 73.0 * k as f64,
+            sat_cost: 1.0
+                + match Ctx::sig_type() {
+                    SigType::Ecdsa => 73.0 * k as f64,
+                    SigType::Schnorr => 64. * k as f64,
+                },
```

Comment:
You would need a `from_multi_a` here. Add appropriate `unreachable!` depending on context

Reply:
Addressed in 2565f5e.

Reply:
I think the implementation of `from_multi_a` is missing here for `CompilerExtData`. Note that the `multi_a` fragment operates differently from `multi` as it consumes empty signatures for dissatisfactions.